cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -844,7 +850,8 @@ public ApplicationWithPrograms deployAppAndArtifact(NamespaceId namespace,
       @Nullable String configStr,
       @Nullable KerberosPrincipalId ownerPrincipal,
       ProgramTerminator programTerminator,
-      boolean updateSchedules) throws Exception {
+      boolean updateSchedules,
+      @Nullable String requestingUserId) throws Exception {
```

Comment:
What is the reason to pass it as parameter? This service has access to AuthenticationContext and can figure it out by itself.

Reply:
This is what I also thought but netty thread keeps on changing hence, this information is lost when it reaches the point of storing information because it `authenticationContext` reads from `SecurityRequestContext` which is maintained per thread: https://github.com/cdapio/cdap/blob/e81bb233fa54e757ace4bbd36a91eec48d967675/cdap-security/src/main/java/io/cdap/cdap/security/auth/context/SystemAuthenticationContext.java#L68

refer to screenshot:
![image](https://github.com/user-attachments/assets/fa6404c7-32cc-4b08-bee2-385881c7fe61)

Reply:
I don't think it's a reason to pass it around. It's a reason to do netty things right. Otherwise other places may have issues

Reply:
Thanks Vitalii for the suggestion, updated the PR as discussed offline.